src/track/beatstats.cpp

```diff
@@ -0,0 +1,55 @@
+#include <QList>
+#include <QMap>
+
+#include <algorithm>
+#include <cmath>
+
+#include "track/beatstats.h"
+
+
+void MovingMode::update(double newValue, double oldValue) {
+    m_tempoFrequency[newValue] += 1;
+    if (not std::isnan(oldValue)) {
+        m_tempoFrequency[oldValue] -= 1;
+    }
+}
+double MovingMode::compute() {
+    QMapIterator<double,  int> tempo(m_tempoFrequency);
+    auto mode = tempo.next();
+    while (tempo.hasNext()) {
+        if (mode.value() < tempo.next().value()) {
```

Comment:
Do not manipulate a variable inside an `if` condition. I think this is why @daschuer was confused. Move `tempo.next()` to its own line.